### PR TITLE
Upgraded to use javalin 6

### LINKED ISSUE #7982: Update Javalin to version 6
Javalin has been updated to a new major version 6

Migration guide: https://javalin.io/migration-guide-javalin-5-to-6

### PR DESCRIPTION
**Pull request by rolfyone**
fixes #7982

I've connected to the rest-api and keymanager-api swagger, fixed the auth access for keymanager.

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/AuthorizationManager.java [hunk 1]
```diff
@@ -16,22 +16,18 @@
...
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.Set;
 import java.util.stream.Stream;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import tech.pegasys.teku.infrastructure.exceptions.InvalidConfigurationException;
 
-public class AuthorizationManager implements AccessManager {
```
Comment (StefanBratanov) [inline infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/AuthorizationManager.java:34]: I think it's cleaner to create `AuthorizationHandler` which implements `Handler` (also to retain the error messages) such as https://gist.github.com/StefanBratanov/2906e76016e59fe59da923df9a1e316d

Then can do `app.beforeMatched(new AuthorizationHandler(path))`

Comment (rolfyone) [reply infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/AuthorizationManager.java:34]: can do, i was just fixing the pattern we had... clearly its not breaking error messages this way... or we've missed a ton of testing...

Comment (rolfyone) [reply infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/AuthorizationManager.java:34]: the context was still being set previously which is what does the error codes etc... even with that gist suggested there substantial changes for anything resembling our tests... just fyi - it wont look super similar.

Comment (rolfyone) [reply infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/AuthorizationManager.java:34]: fixed now :)

File: infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/AuthorizationHandler.java [hunk 2]
```diff
@@ -88,13 +85,12 @@ private void unauthorized(final Context ctx, final String message) {
     LOG.debug(message);
     ctx.json("{\n  \"status\": 401, \n \"message\":\"Unauthorized\"\n}");
     ctx.status(SC_UNAUTHORIZED);
+    ctx.skipRemainingHandlers();
```
Comment (StefanBratanov) [inline infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/AuthorizationHandler.java:88]: That's the magic I suppose

File: infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/RestApiBuilder.java [hunk 3]
```diff
@@ -182,16 +184,15 @@ private void addExceptionHandler(
...
-            cors -> corsAllowedOrigins.forEach(origin -> cors.add(it -> it.allowHost(origin))));
+        config.bundledPlugins.enableCors(
+            cors -> corsAllowedOrigins.forEach(origin -> cors.addRule(it -> it.allowHost(origin))));
       }
     }
   }
 
-  private Server createJettyServer() {
-    final Server server = new Server();
+  private void modifyJettyServer(Server server) {
```
Comment (StefanBratanov) [inline infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/RestApiBuilder.java]: final 🗡️

**Review by StefanBratanov — APPROVED** [review]
LGTM (apart from one final haha)

File: infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/SwaggerUIBuilder.java [hunk 4]
```diff
@@ -84,4 +90,20 @@ public Optional<String> configureDocs(
...
   }
+
+  private JavalinThymeleaf createThymeleafRenderer(final String templatePath) {
+    TemplateEngine templateEngine = new TemplateEngine();
+    templateEngine.addTemplateResolver(templateResolver(templatePath));
+    return new JavalinThymeleaf(templateEngine);
+  }
+
+  private ITemplateResolver templateResolver(final String prefix) {
+    ClassLoaderTemplateResolver templateResolver =
```
Comment (rolfyone) [inline infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/SwaggerUIBuilder.java]: @StefanBratanov you missed one 🗡️

**Pull request merged**